### PR TITLE
[FIX] mail: portal user cannot use public discuss

### PR DESCRIPTION
**Pull request by zel-odoo**

### CODE REVIEW

File: addons/mail/controllers/discuss.py [hunk 1]
```diff
@@ -192,7 +192,9 @@ def fetch_image(self, channel_id, attachment_id, width=0, height=0, **kwargs):
     @http.route('/mail/init_messaging', methods=['POST'], type='json', auth='public')
     def mail_init_messaging(self, **kwargs):
         if not request.env.user.sudo()._is_public():
-            return request.env.user.sudo(False)._init_messaging()
+            if request.env.user._is_internal():
+                return request.env.user.sudo(False)._init_messaging()
+            return request.env.user._init_messaging()
```
Comment (seb-odoo) [inline addons/mail/controllers/discuss.py:197]: This change should be done in master directly, and have a test to cover it.

Also I think this change might be too naive and leak potentially unwanted information to portal users.

Comment (seb-odoo) [reply addons/mail/controllers/discuss.py:197]: Not master actually, but v15? v16? whatever is the first version where it fails.

**Review by alexkuhn — COMMENTED** [review]
The commit/PR title is misleading from the changes you make.
It looks like the fix is to allow portal user to use public discuss.

**Comment by zel-odoo** [comment]
> The commit/PR title is misleading from the changes you make. It looks like the fix is to allow portal user to use public discuss.

It is indeed when

**Pull request closed without merging**